File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/MissingDependencyNotification.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getAndReplaceText;
import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
import org.jetbrains.annotations.NotNull;

public class MissingDependencyNotification extends Notification {
  /**
   * Construct a missing module notification that explains that a module with the given name
   * couldn't be found.
   */
  public MissingDependencyNotification(@NotNull String moduleNames) {
    super(
        PluginSettings.A_PLUS,
        getText("notification.MissingDependencyNotification.title"),
        getAndReplaceText("notification.MissingDependencyNotification.content", moduleNames),
        NotificationType.ERROR);
  }

}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/MissingDependencyNotification.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

/**
 * Explains that a module declares dependencies on modules the course does not provide, so they
 * cannot be installed automatically.
 *
 * Raised by CourseManager when refreshing module statuses.
 */
class MissingDependencyNotification(moduleNames: Set<String>) : Notification(
    PluginSettings.A_PLUS,
    MyBundle.message("notification.MissingDependencyNotification.title"),
    MyBundle.message(
        "notification.MissingDependencyNotification.content",
        moduleNames.joinToString(", ")
    ),
    NotificationType.ERROR
)
